p2p/module.go

```diff
@@ -0,0 +1,193 @@
+package p2p
+
+import (
+	"errors"
+	"log"
+	"net"
+	"pocket/p2p/types"
+	"pocket/shared/config"
+	"pocket/shared/modules"
+	"pocket/shared/types"
+	"sync"
+
+	"go.uber.org/atomic"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+var (
+	MaxInbound           uint = 128
+	MaxOutbound          uint = 128
+	WireByteHeaderLength int  = 9
+	ReadBufferSize       int  = (1024 * 4)
+	WriteBufferSize      int  = (1024 * 4)
+	ReadDeadlineMs       int  = 400
+)
+
+var (
+	Protocol = "tcp"
+	Address  = "localhost:30303"
+)
+
+type P2PModule struct {
+	modules.NetworkModule
+
+	bus    modules.Bus
+	config *config.P2PConfig
+	c      *dcodec
+
+	id           uint64
+	protocol     string
+	address      string
+	externaladdr string
+
+	inbound  pipemap
+	outbound pipemap
```

Comment:
Consider making the naming more descriptive. E.g. `outboundNetworkPipe`. Doesn't need to be super verbose, but maybe a comment could help too for the reader.

Reply:
To be honest, I want to rename `pipe` into `socket`, but I wonder if that's too confusing... In which case it will be somewhat clear that `outbound socketmap` means it's a map of currently opened outbound connections.